File: solutions/armstrong-numbers/Python/armstrong_numbers.py
from math import log10
def is_armstrong_number(number):
    if number < 0: return False
    if number == 0: return True
    n_digits = int(log10(number)) + 1
    return number == sum(((number // 10**k % 10)**n_digits for k in range(n_digits)))

File: solutions/armstrong-numbers/C++/armstrong_numbers.cpp
#include "armstrong_numbers.h"

bool armstrong_numbers::is_armstrong_number(const int n) noexcept {
    const int order = (n <= 0)? 0: std::log10(n) + 1;
    int sum = 0;
    for (int i = n; i > 0; i /= 10)
        sum += std::pow(i % 10, order);
    return sum == n;
}